velox/functions/prestosql/GeometryUtils.h

```diff
@@ -0,0 +1,685 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <geos/geom/GeometryFactory.h>
+#include <geos/geom/LineString.h>
+#include <geos/geom/Point.h>
+
+#include <velox/common/base/Exceptions.h>
+#include <velox/type/StringView.h>
+
+namespace facebook::velox::functions {
+
+class GeometryUtils {
+ public:
+  template <typename StringWriter>
+  static void serialize(
+      const std::unique_ptr<geos::geom::Geometry>& geometry,
```

Comment:
nit: not sure if its ok  to pass a const ref to a unique_ptr and then pass the raw ptr to serialize ? (maybe shared if geometry is to be reused or move it ?)